tools/serverpod_cli/lib/src/generator/dart/library_generators/util/model_filter_util.dart

```diff
@@ -0,0 +1,56 @@
+import 'package:path/path.dart' as p;
+
+import 'package:serverpod_cli/src/analyzer/models/definitions.dart';
+import 'package:serverpod_cli/src/config/config.dart';
+
+List<ClassDefinition> getSealedClasses(
+  List<SerializableModelDefinition> models,
+) {
+  return models
+      .whereType<ClassDefinition>()
+      .where((element) => element.sealedTopNode == element)
+      .toList();
+}
+
+List<List<ClassDefinition>> getSealedHierarchies(
+  List<SerializableModelDefinition> models,
+) {
+  var sealedClasses = getSealedClasses(models);
+
+  return sealedClasses.map(
+    (element) {
+      return [...element.descendantClasses, element];
+    },
+  ).toList();
+}
+
+List<ClassDefinition> getSealedHierarchyClasses(
+  List<SerializableModelDefinition> models,
+) {
+  return getSealedHierarchies(models).expand((e) => e).toList();
+}
+
+List<SerializableModelDefinition> getClassesWithoutSealedHierarchies(
+  List<SerializableModelDefinition> models,
+) {
+  return models
+      .where((e) => !getSealedHierarchyClasses(models).contains(e))
```

Comment:
Suggestion: Change this to only loop the list 1 on 2 times.

Since `getSealedHierachyClasses` loops the full list each time it is invoked, this will loop the list for every model in `models`.
Figure out what are the sealed classes first and then check if they contain the entry.